### PR TITLE
Implemented decoder generation from isa.yaml

### PR DESCRIPTION
**Pull request by MrYar0s**
Now this PR without tests
Need refactoring and unit tests

### CODE REVIEW

**Comment by MrYar0s** [comment]
PR also don't has tests, but I refactored code using rubocop and added some common bitops functions, so you can start your review

File: common/bitops.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	return raw_inst >> Shift;
+}
+
+template <uint32_t Mask, int8_t From, int8_t To>
+uint32_t ApplyMaskAndShift(uint32_t raw_inst)
+{
+	static_assert(From > 0 && To > 0);
+	static_assert(From < sizeof(uint32_t)*CHAR_BIT && To < sizeof(uint32_t)*CHAR_BIT);
+	raw_inst = ApplyMask<uint32_t, Mask>(raw_inst);
+	constexpr int8_t diff = From - To;
```
Comment (PozhiloyPumba) [inline common/bitops.h:35]: why can't we use std::max(From -To, To - From); and static_assert(From != To);

File: common/bitops.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#ifndef COMMON_BITOPS_H
+#define COMMON_BITOPS_H
+
+#include <cassert>
+#include <climits>
+
+namespace sim {
```
Comment (Mirageinvo) [inline common/bitops.h:7]: Please, rename namespace to 'simulator' or change namespace name to 'sim' in memory folder

File: interpreter/decoder.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#ifndef INTERPRETER_DECODER_H
+#define INTERPRETER_DECODER_H
+
+#include "interpreter/instruction.h"
+
+namespace sim {
```
Comment (Mirageinvo) [inline interpreter/decoder.h:6]: Please, rename namespace to 'simulator' or change namespace name to 'sim' in memory folder

**Review by PozhiloyPumba — APPROVED** [review]

**Review by Mirageinvo — APPROVED** [review]

**Pull request merged**